jax/scipy/special.py

```diff
@@ -101,15 +101,26 @@ def expit(x):
 
 @_wraps(osp_special.logsumexp)
 def logsumexp(a, axis=None, b=None, keepdims=False, return_sign=False):
-  if b is not None or return_sign:
-    raise NotImplementedError("Only implemented for b=None, return_sign=False")
+  if b is not None:
+    a, b = jnp.broadcast_arrays(a, b)
   dims = _reduction_dims(a, axis)
   dimadd = lambda x: lax.expand_dims(x, dims)
   amax = lax.reduce(a, _constant_like(a, -np.inf), lax.max, dims)
   amax = lax.stop_gradient(lax.select(lax.is_finite(amax), amax, lax.full_like(amax, 0)))
   amax_singletons = dimadd(amax)
-  out = lax.add(lax.log(lax.reduce(lax.exp(lax.sub(a, amax_singletons)),
-                                   _constant_like(a, 0), lax.add, dims)), amax)
+  if b is None:
+    out = lax.add(lax.log(lax.reduce(lax.exp(lax.sub(a, amax_singletons)),
+                                     _constant_like(a, 0), lax.add, dims)), amax)
+    sign = lax.stop_gradient(lax.sign(out))
```

Comment:
Shoot, given the test failures I think I misled you here. We need the sign of `exp(out)`, not the sign of `out`. I think this should probably be something like this:
```
sign = jnp.where(jnp.isnan(out), np.nan, 1.0)
```